tests/qnn/test_keras.py

```diff
@@ -34,24 +34,44 @@ def model(get_circuit, n_qubits, output_dim):
     layer1 = KerasLayer(c, w, output_dim)
     layer2 = KerasLayer(c, w, output_dim)
 
-    model = tf.keras.models.Sequential(
-        [
-            tf.keras.layers.Dense(n_qubits),
-            layer1,
-            tf.keras.layers.Dense(n_qubits),
-            layer2,
-            tf.keras.layers.Dense(output_dim),
-        ]
-    )
+    if isinstance(output_dim, tuple):
+
+        model = tf.keras.models.Sequential()
+        model.add(tf.keras.layers.Dense(n_qubits))
+        model.add(layer1)
+        model.add(tf.keras.layers.Flatten())
+        model.add(tf.keras.layers.Dense(n_qubits))
+        model.add(layer2)
+        model.add(tf.keras.layers.Flatten())
+        model.add(tf.keras.layers.Dense(output_dim[0] * output_dim[1]))
+
+    else:
+        model = tf.keras.models.Sequential(
+            [
+                tf.keras.layers.Dense(n_qubits),
+                layer1,
+                tf.keras.layers.Dense(n_qubits),
+                layer2,
+                tf.keras.layers.Dense(output_dim),
+            ]
+        )
 
     return model
 
 
-def indices_up_to(n_max):
+def indices_up_to(n_max, has_tuple=False):
     """Returns an iterator over the number of qubits and output dimension, up to value n_max.
     The output dimension never exceeds the number of qubits."""
-    a, b = np.tril_indices(n_max)
-    return zip(*[a + 1, b + 1])
+    if has_tuple:
+        # If the output_dim is to be used as a tuple. First element is for n_qubits and
+        # the second is for output_dim. For example, for n_max = 3 it will return,
+        # [(1, (2, 2)), (2, (2, 2)), (3, (2, 2)), (3, (3, 3))]
```

Comment:
```suggestion
        # [(1, (2, 2)), (2, (2, 2)), (2, (4, 4)), (3, (2, 2)), (3, (4, 4)), (3, (8, 8))]
```

Reply:
Thanks for catching that! I must've missed it.